src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,401 @@
+                var normalizedRight = NormalizeOSName(right);
+                return string.Equals(normalizedLeft, normalizedRight, StringComparison.OrdinalIgnoreCase);
+            }
+        }
+
+        private static string NormalizeOSName(string name) => name.Equals(OSX, StringComparison.OrdinalIgnoreCase)
+            ? macOS
+            : name;
+
+        private static DiagnosticDescriptor SelectRule(PlatformAttributeType attributeType)
+            => attributeType switch
+            {
+                PlatformAttributeType.MinimumOSPlatformAttribute => MinimumOsRule,
+                PlatformAttributeType.ObsoletedInOSPlatformAttribute => ObsoleteRule,
+                _ => RemovedRule,
+            };
+
+        private static ImmutableArray<AttributeData> FindAllPlatformAttributesApplied(ImmutableArray<AttributeData> immediateAttributes, INamedTypeSymbol parent)
+        {
+            using var builder = ArrayBuilder<AttributeData>.GetInstance();
+            foreach (AttributeData attribute in immediateAttributes)
+            {
+                if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name))
+                {
+                    builder.Add(attribute);
+                }
+            }
+
+            while (parent != null)
+            {
+                var current = parent.GetAttributes();
+                foreach (var attribute in current)
+                {
+                    if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name) &&
+                        TargetPlatformAttribute != attribute.AttributeClass.Name)
+                    {
+                        builder.Add(attribute);
+                    }
+                }
+                parent = parent.BaseType;
+            }
+
+            return builder.ToImmutableArray();
+        }
+
+        private static bool IsSuppressedByAttribute(PlatformAttributeInfo diagnosingAttribute, ISymbol containingSymbol)
+        {
+            while (containingSymbol != null)
+            {
+                var attributes = containingSymbol.GetAttributes();
+                if (!attributes.IsEmpty)
+                {
+                    foreach (AttributeData attribute in attributes)
+                    {
+                        if (diagnosingAttribute.AttributeType.ToString() == attribute.AttributeClass.Name)
+                        {
+                            if (PlatformAttributeInfo.TryParsePlatformAttributeInfo(attribute, out PlatformAttributeInfo parsedAttribute))
+                            {
+                                if (IsOSPlatformsEqual(diagnosingAttribute.PlatformName, parsedAttribute.PlatformName) &&
+                                    AttributeVersionsMatch(diagnosingAttribute, parsedAttribute))
+                                {
+                                    return true;
+                                }
+                            }
+                        }
```

Comment:
```suggestion
                        if (diagnosingAttribute.AttributeType.ToString() == attribute.AttributeClass.Name &&
                            PlatformAttributeInfo.TryParsePlatformAttributeInfo(attribute, out PlatformAttributeInfo parsedAttribute) &&
                            IsOSPlatformsEqual(diagnosingAttribute.PlatformName, parsedAttribute.PlatformName))
                        {
                            return AttributeVersionsMatch(diagnosingAttribute, parsedAttribute);
                        }
```

Reply:
I am OK with moving all conditions into one, but not moving the last one into 
`return AttributeVersionsMatch(diagnosingAttribute, parsedAttribute);` 
as if it false we shouldn't return, need to continue checking

Reply:
Why is that required? If I found a containing symbol with matching OS platform and the version on it is satisfied, why would be continue walking further up? Can you add verify the current behavior matches expected behavior for the below test?

```csharp
[MinimumOSPlatform("Windows10")]
class C
{
    [MinimumOSPlatform("Windows11")]
    void Windows11Method()
    {
        AnotherWindows11Method(); // This should not be flagged, but will be flagged with current algorithm?
    }

    [MinimumOSPlatform("Windows11")]
    void AnotherWindows11Method()
    {
    }

    [MinimumOSPlatform("Windows10")]
    void Windows10Method()
    {
    }
}
```

Reply:
That will not be flagged because of the attribute on the  `Windows11Method`, versions match so it will return true, it is not the opposite case. But the scenario you are talking could be shown with the last method i think:
```csharp
    [MinimumOSPlatform("Windows10")]
    void Windows10Method()
    {
          AnotherWindows11Method();
    }
```
Here version is less so by my approach it should check other attributes of containing symbol